libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -265,6 +279,22 @@ void _usage(void)
 
 void HAL_Linux::run(int argc, char* const argv[], Callbacks* callbacks) const
 {
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_IMX
+    //some times ardupilot bootup,but cmd->"modprobe cdc_acm" create file node->"/dev/ttyACM0" Not yet!
+    //just dead loop until file node created.
+    while (1)
```

Comment:
what kind of init system are you using? This should be a dependency on starting the ardupilot  process

Reply:
imx6 chip use usb cdc_acm connect to stm32 chip,stm32 use for send pwm.
so before start ardupilot process, i must insert a kernel module driver "cdc_acm.ko",
use command "modprobe cdc_acm"

Reply:
and what linux distro are you using? The init is probably either systemd or sysv. Any of them allow you to call modprobe just before starting ardupilot.

Reply:
i use ubuntu base 14.04,but i use busybox to init the system.
i can not use ubuntu base directly in this board,it get Bad performance.
just use busybox init system will get perfect performance;

Give me more time to test it

Reply:
ok. even busybox's init should be able to do it. I guess you are adding ardupilot to be started from /etc/init.d/rcS ?

just add a line to modprobe the module.

I'm not sure about the the ordering in that file for you, but adding the module to /etc/modules may fix it.